src/utils/url.js

```diff
@@ -42,11 +48,15 @@ export function urlParams (url) {
   }
   queryParams.split('&').forEach(function (query) {
     if (query) {
-      query = ((query = query.split('=')) && query.length === 2 ? query : [query[0], 'true']).map(_decode)
+      query = ((query = query.split('=')) && query.length === 2 ? _queryNameValue(query, predicate) : [query[0], 'true'])
```

Comment:
This will only decode the parameter when the predicate matches and return the original value otherwise. 

I actually meant that we can create a function that allows to extract only one particular parameter (`getQueryParameter(url, name)`). And the code would also attempt to only decode this one parameter, do not process other parameters at all and fail if it is not possible. The function can then be is used [here](https://github.com/LiveIntent/live-connect/blob/master/src/manager/decisions.js#L31-L32) to extract lidids.

Reply:
I agree that reading a single parameter is probably the way to go.

An alternative would be to go the same way as our cookie library and manually only decode the urlencoded parts.
`value.replace(/(%[\dA-F]{2})+/gi, decodeURIComponent)`
That is less brittle than our approach.

source: https://github.com/js-cookie/js-cookie/blob/main/src/converter.mjs#L7

Reply:
when I try it with this value 
```
'https://www.bluenile.com/hk/zh/diamonds?gclid=CjwKCAiAh9q&click_id=12&utm_source=google&utm_campaign=Google_%7C_GC_%7C_HK_%7C_Traditional_Chinese_%7C_Text_%7C_Non-Brand_%7C_ENG_%7C_NA_%7C_Engagement_%7&utm_content=Diamonds%3A_Price&utm_term=%E4%B8%80+%E5%85%8B%E6%'
```
it will return the same error:
```
Uncaught URIError: URI malformed
```

Reply:
Good suggestion though
I tried the `getQueryParameter` solution